src/module/chat-message/listeners/cards.ts

```diff
@@ -233,14 +235,24 @@ class ChatCards {
                 throw ErrorPF2e(`"${saveType}" is not a recognized save type`);
             }
 
-            const dc = Number(button.dataset.dc ?? "NaN");
+            const localize = localizer("PF2E.InlineCheck");
+            const dc = ((): CheckDC | null => {
+                const v = Number(button.dataset.dc ?? "NaN");
+                if (Number.isInteger(v)) {
+                    const name = item?.name;
+                    const label = localize("DCWithName", { name });
+                    return { label, value: v };
+                }
+                return null;
+            })();
+
```

Comment:
This entire block looks like a separate feature that should probably be supported in Statistic#roll(), should we choose to handle it. This affects spells as well, and should probably be for everything if we want it.

If you think it is indeed better, perhaps separating this into a new thing handled by Statistic's roll function might be best.